src/xdist/plugin.py

```diff
@@ -1,28 +1,13 @@
-import os
 import uuid
 
+import psutil
 import py
 import pytest
 
 
 def auto_detect_cpus():
     try:
-        from os import sched_getaffinity
-    except ImportError:
-        if os.environ.get("TRAVIS") == "true":
-            # workaround https://bitbucket.org/pypy/pypy/issues/2375
-            return 2
-        try:
-            from os import cpu_count
-        except ImportError:
-            from multiprocessing import cpu_count
-    else:
-
-        def cpu_count():
-            return len(sched_getaffinity(0))
-
-    try:
-        n = cpu_count()
+        n = psutil.cpu_count(logical=False)
     except NotImplementedError:
```

Comment:
is this something that psutil will raise?

> Return the number of logical CPUs in the system (same as os.cpu_count in Python 3.4) or None if undetermined. logical cores means the number of physical cores multiplied by the number of threads that can run on each core (this is known as Hyper Threading). If logical is False return the number of physical cores only (Hyper Thread CPUs are excluded) or None if undetermined. On OpenBSD and NetBSD psutil.cpu_count(logical=False) always return None. Example on a system having 2 physical hyper-thread CPU cores:

if it never raises, we can use `return psutil.cpu_count(logical=False) or 1`

Reply:
I'm also concerned about:
> On OpenBSD and NetBSD psutil.cpu_count(logical=False) always return None.

should we fallback to the old code if we get a `None`?

Reply:
> should we fallback to the old code if we get a None?

Not sure, we don't officially support OpenBSD and NetBSD... your suggestion above of using `return psutil.cpu_count(logical=False) or 1` would at least prevent a crash, and I would rather keep the code cleaner. But if others disagree then I think your suggestion is on point.

Reply:
Does this obviate the Travis workaround?

Reply:
Good question. 👍

Reply:
![image](https://user-images.githubusercontent.com/1124655/88611014-552d2700-d0b2-11ea-80a7-2eaa7e14687a.png)
I tested this locally on pypy35.